common/buildcraft/lib/misc/BoxIterator.java

```diff
@@ -92,4 +137,19 @@ private boolean shouldReset(BlockPos current, EnumFacing facing) {
     public boolean hasFinished() {
         return current == null;
     }
+
+    @Override
+    public Object readFromByteBuf(ByteBuf buf) {
+        return new BoxIterator(NetworkUtils.readBlockPos(buf), NetworkUtils.readBlockPos(buf), buf.readBoolean(), buf.readBoolean(), (AxisOrder) new AxisOrder().readFromByteBuf(buf), NetworkUtils.readBlockPos(buf));
+    }
+
+    @Override
+    public void writeToByteBuf(ByteBuf buf) {
+        NetworkUtils.writeBlockPos(buf, min);
+        NetworkUtils.writeBlockPos(buf, max);
+        buf.writeBoolean(invert);
+        buf.writeBoolean(repeat);
+        order.writeToByteBuf(buf);
+        NetworkUtils.writeBlockPos(buf, current);
+    }
```

Comment:
It would probably be better to calculate the progress on the server and send over the single int rather than sending over the entire box iterator (so can you remove the read/write methods from this class)